core-bundle/src/Filesystem/PublicUri/PublicUriProviderInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Filesystem\PublicUri;
+
+use League\Flysystem\FilesystemAdapter;
+use Psr\Http\Message\UriInterface;
+
+interface PublicUriProviderInterface
+{
+    /**
+     * Return a public URI for this resource or null if this provider does not
+     * support the given adapter/path/options.
+     */
+    public function getUri(FilesystemAdapter $adapter, string $adapterPath, Options $options): ?UriInterface;
```

Comment:
See my other comment, maybe `Options` should be an interface here.

Reply:
see 31b535260b06297e14e8fbb45f8df92ebb05d3f5

Default settings (or maybe arbitrary ones in an array) are still to be figured out. :slightly_smiling_face: